src/jobs/helm_aws.yml

```diff
@@ -106,7 +110,18 @@ steps:
             cmd_name: Reset the replica count of each deployment for the specified env
             deploy: false
             cmd: |
-              if [[ "<< parameters.reset_dev_env_replica_count >>" =~ ^dev-env.* ]]; then
+              if ! [[ -z "<< parameters.scale_down_time_limit >>" ]]; then
+                current_timestamp=$(date +%s)
+                time_limit_timestamp=$(date -d "<< parameters.scale_down_time_limit >>" +%s)
+                if [ "$current_timestamp" -gt "$time_limit_timestamp" ]; then
+                  echo "Current time is after after the specified time limit. Exiting..."
+                  exit 1
```

Comment:
Maybe in a future revision it would make sense to make exiting on error optional, since in some scenarios it could be ok and intended to just exit 0 from this step and proceed to the next one(s).

Reply:
Note taken!